### PR TITLE
logging of stance changes + end game stance flip

### PR DESCRIPTION
**Pull request by RunzheYang**
import new stance module;
stance logging via game log

### CODE REVIEW

File: src/baseline_bots/bots/smart_order_accepter_bot.py [hunk 1]
```diff
@@ -280,6 +280,13 @@ async def respond_to_invalid_orders(
             )
             await self.send_message(sender, str(message), messages_data)
 
+    async def log_stance_change(self, stance_log) -> None:
+        for pw in list(self.game.get_map_power_names()):
+            if pw == self.power_name:
+                continue
+            log_data = self.game.new_log_data(body=stance_log[self.power_name][pw])
```
Comment (aphedges) [inline src/baseline_bots/bots/smart_order_accepter_bot.py:287]: This kind of loop is common enough that I created a field to store a list of all other powers:

```suggestion
        for pw in self.opponents:
            log_data = self.game.new_log_data(body=stance_log[self.power_name][pw])
```

**Review by aphedges — CHANGES_REQUESTED** [review]
Once you've made this change, I'll try it out!

**Review by aphedges — CHANGES_REQUESTED** [review]
I can confirm this works properly! Because intent logging is not visible in the UI, I created a cleaner JSON file from the game state using the command `jq '[.phases[] | {name, logs}]'`: [ahedges_2023_02_26_00_36_32_intent_log.json.zip](https://github.com/ALLAN-DIP/baseline_bots/files/10832640/ahedges_2023_02_26_00_36_32_intent_log.json.zip)

Now, this branch just needs to be cleaned up before it can be merged. Here are the steps you should follow:

```bash
git fetch # To make sure local repo is up to date
git rebase origin/main # To remove unneeded merge commit
git rebase -i HEAD~5 # To squash all commits together
# Change `pick` to `f` in all rows but the first, then save the file
git commit --amend # To change commit message
# Update commit message, then save the file
git commit --force-with-lease # Push changes to GitHub
```

For the changed commit message when running `git commit --amend`, I suggest something like the following:

```text
Add intent logging for stance changes to SOA bot

This update to `stance_vector` also adds the betrayal coin flip
behavior.
```

**Comment by RunzheYang** [comment]
Thanks! I cleaned up the branch!

**Review by aphedges — APPROVED** [review]
Looks all good to me!

**Pull request merged**